test/unit/sync_checker/checks/details_check_test.rb

```diff
@@ -0,0 +1,145 @@
+require 'minitest/autorun'
+require 'mocha/setup'
+require 'active_support'
+require 'active_support/json'
+require 'active_support/core_ext'
+
+require_relative '../../../../lib/sync_checker/checks/details_check'
+
+module SyncChecker::Checks
+  class DetailsCheckTest < Minitest::Test
+    def test_non_200_returns_an_empty_array
+      response = stub(
+        response_code: 404,
+        body: {
+        }.to_json
+      )
+
+      expected = {
+        body: "a bit of html"
+      }
+
+      assert_equal [], DetailsCheck.new(expected).call(response)
```

Comment:
Do 404s and other unexpected responses get reported elsewhere? Is this considered different to not in sync or is it just noise you want to omit?